src/Image_Backend.php

```diff
@@ -84,87 +78,91 @@ public function setImageResource($resource);
-     *
-     * @param int $width
-     * @param int $height
-     * @param string $backgroundColor
-     * @param int $transparencyPercent
-     * @return static
+     * Return a clone of this image resized, with space filled in with the given colour.
      */
-    public function paddedResize($width, $height, $backgroundColor = "FFFFFF", $transparencyPercent = 0);
+    public function paddedResize(string $width, string $height, string $backgroundColour = 'FFFFFF', int $transparencyPercent = 0): ?static;
 
     /**
      * Resize an image to cover the given width/height completely, and crop off any overhanging edges.
-     *
-     * @param int $width
-     * @param int $height
-     * @return static
      */
-    public function croppedResize($width, $height);
+    public function croppedResize(int $width, int $height, string $position = 'center'): ?static;
 
     /**
      * Crop's part of image.
-     * @param int $top y position of left upper corner of crop rectangle
-     * @param int $left x position of left upper corner of crop rectangle
+     * @param int $top Amount of pixels the cutout will be moved on the y (vertical) axis
+     * @param int $left Amount of pixels the cutout will be moved on the x (horizontal) axis
      * @param int $width rectangle width
      * @param int $height rectangle height
-     * @return Image_Backend
+     * @param string $position Postion at which the cutout will be aligned
+     * @param string $backgroundColour Colour to fill any newly created areas
+     */
+    public function crop(int $top, int $left, int $width, int $height, string $position, string $backgroundColour = 'FFFFFF'): ?static;
+
+    /**
+     * Set whether this image backend is allowed to output animated images as a result of manipulations.
+     */
+    public function setAllowsAnimationInManipulations(bool $allow): static;
+
+    /**
+     * Get whether this image backend is allowed to output animated images as a result of manipulations.
+     */
+    public function getAllowsAnimationInManipulations(): bool;
+
+    /**
+     * Check if the image is animated (e.g. an animated GIF).
+     */
+    public function getIsAnimated(): bool;
+
+    /**
+     * Discards all animation frames of the current image instance except the one at the given position. Turns an animated image into a static one.
+     *
+     * @param integer|string $position Which frame to use as the still image.
+     * If an integer is passed, it represents the exact frame number to use (starting at 0). If that frame doesn't exist, an exception is thrown.
+     * If a string is passed, it must be in the form of a percentage (e.g. '0%' or '50%'). The frame to use is then determined based
+     * on this percentage (e.g. if '50%' is passed, a frame halfway through the animation is used).
      */
-    public function crop($top, $left, $width, $height);
+    public function removeAnimation(int|string $position): ?static;
```

Comment:
Note that there are two different but related concepts here.

On one hand, if animation is _not allowed_ then all variants will result in a still frame regardless of whether the original image was animated or not.

On the other hand, if animation _is allowed_, developers can create a variant which is a still frame of the original animated image using `removeAnimation`. This could be useful for getting a _specific_ frame (not just the first frame like when animated variants aren't allowed), or for scenarios where you're generating a series of variants and want some to be animated, and some to be still.